src/Microsoft.IdentityModel.Protocols.MSPop/MSPopTokenCreationData.cs

```diff
@@ -0,0 +1,92 @@
+
+namespace Microsoft.IdentityModel.Protocols.MSPop
+{
+    /// <summary>
+    /// Structure that wraps parameters needed for MSPop token creation. 
+    /// </summary>
+    public class MSPopTokenCreationData : MSPopTokenData
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="MSPopTokenCreationData"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// Default <see cref="CallContext"/> is created.
+        /// </remarks>
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="msPopTokenSigningCredentials">A security key and algorithm that will be used to sign the MSPop token.</param>
+        /// <param name="msPopTokenCreationPolicy">A policy for creating the MSPop token.</param>
+        public MSPopTokenCreationData(string accessToken, HttpRequestData httpRequestData, SigningCredentials msPopTokenSigningCredentials, MSPopTokenCreationPolicy msPopTokenCreationPolicy) 
+            : base(httpRequestData, CallContext.Default)
+        {
+            AccessToken = !string.IsNullOrEmpty(accessToken) ? accessToken : throw LogHelper.LogArgumentNullException(nameof(accessToken));
+            MSPopTokenSigningCredentials = msPopTokenSigningCredentials ?? throw LogHelper.LogArgumentNullException(nameof(msPopTokenSigningCredentials));
+            MSPopTokenCreationPolicy = msPopTokenCreationPolicy ?? throw LogHelper.LogArgumentNullException(nameof(msPopTokenCreationPolicy));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MSPopTokenCreationData"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// </remarks> 
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="msPopTokenSigningCredentials">Defines the security key and algorithm that will be used to sign the MSPop token.</param>
+        /// <param name="msPopTokenCreationPolicy">A policy for creating the MSPop token.</param>
+        /// <param name="callContext">An opaque context used to store work when working with authentication artifacts.</param> 
+        public MSPopTokenCreationData(string accessToken, HttpRequestData httpRequestData, SigningCredentials msPopTokenSigningCredentials, MSPopTokenCreationPolicy msPopTokenCreationPolicy, CallContext callContext) 
```

Comment:
did you consider the name PopTokenDescriptor, this would be in line with SecurityTokenDescriptor.
it has similar semantics.